src/components/form/ProxyTypeSelect.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import CustomDropdown from '../shared/CustomDropdown';
+import type { DropdownOption } from '../shared/CustomDropdown';
+
+interface ProxyTypeSelectProps {
+  value: string;
+  onChange: (value: string) => void;
+  isDisabled?: boolean;
+}
+
+const PROXY_OPTIONS: DropdownOption[] = [
+  { label: 'Public', value: 'public' },
+  { label: 'Private', value: 'private' },
+];
+
+const ProxyTypeSelect: React.FC<ProxyTypeSelectProps> = ({
```

Comment:
I think it's better to remove component type as a generic for code base integrity.
```typescript
const ProxyTypeSelect = ({
  value,
  onChange,
  isDisabled,
}: ProxyTypeSelectProps)
```